### ISSUE TITLE
Infinity scroll pagination question

### ISSUE THREAD

**Issue by ignl** [ramezrafla/redis-oplog#15]
What I get with this version of redis-oplog is that when I scroll to the next page and reactively update limit parameter subscription is stopped and resubscribed which causes all records to be removed and re-added instead of just adding records for new page. Is it expected?

**Comment by ramezrafla** [comment]
How does your pub look like?

**Comment by ramezrafla** [comment]
I don't have lazy load in my app

**Comment by ignl** [comment]
Simplified version like this:
```
Meteor.publish("posts", function (limit) {
    return posts.find({}, {sort: {'date' : -1}, limit: limit});
})

this.subscribe("posts", Session.get('limit'))
```

When limit variable is increased in client normal behavior is to unsubscribe from old subscription and subscribe to a new one, but meteor just sends out new docs as old ones are already in minimongo. With this redis-oplog I see that all entries in minimongo is removed and everything is sent again which is of course looks bad in client.

**Comment by ramezrafla** [comment]
@ignl 

The fact that your code worked before but not now, seems more like a quirk than intended design of Meteor. We may execute the unsub faster than original Meteor.

1. Do you sub to the new then unsub? This should fix things, and prevent server-side pub from resending docs. You are benefiting from the mergebox
2. I think the right and clean approach is to send a `skip` option to the pub. Keep subscribing to new pubs as you need to, which only sends new docs. When the view is closed, unsub from all of them (store their handlers in an array)

**Comment by ramezrafla** [comment]
```
Meteor.publish("posts", function (limit, skip) {
    return posts.find({}, {sort: {'date' : -1}, limit, skip});
})

const limit = Session.get('limit')
subs.push(this.subscribe("posts", limit, limit ? limit - COUNT : 0))

useEffect(() => () => subs.forEach(s => s.stop()), [])
```

**Comment by ramezrafla** [comment]
[Note that I edited the two comments above, pls view online]

**Comment by ignl** [comment]
> @ignl
> 
> The fact that your code worked before but not now, seems more like a quirk than intended design of Meteor. We may execute the unsub faster than original Meteor.
> 
>     1. Do you sub to the new then unsub? This should fix things, and prevent server-side pub from resending docs. You are benefiting from the mergebox
> 
>     2. I think the right and clean approach is to send a `skip` option to the pub. Keep subscribing to new pubs as you need to, which only sends new docs. When the view is closed, unsub from all of them (store their handlers in an array)

@ramezrafla It was certainly not a quirk and is documented meteor behavior (link [here](https://guide.meteor.com/data-loading.html#publication-behavior-with-arguments)). So this is a common pattern in Meteor and as documented the way Meteor works is not stopping old subscription until the new one is fully ready. Does redis-oplog work different in such case?

**Comment by ramezrafla** [comment]
@ignl
 
That pattern is common, I am referring to the fact that the data persisted a bit AFTER you unsubscribed, which gave you a time to resubscribe again without losing data. If you notice in the doc your are mentioning, there is no unsub. You keep subscribing with a higher limit. 

My points above are still valid. You should have no time where you are not unsubscribed so you don't get flicker.

**Comment by ignl** [comment]
OK, so I checked other changes I did after your comments and found major one I forgot about - doing this.unblock() on publications after update to meteor 2.3, so that was the culprit. Sorry for the false alarm :)
